src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -202,31 +209,29 @@ public async Task DeleteFileIfExistsAsync(long version, Partition partition, Fil
         {
             try
             {
-                bool exists = await blobClient.ExistsAsync(cancellationToken);
-                // todo change this back to DeleteIfExistsAsync when bug fixed story#110757
-                // when file does not exist but conditions passed in, it fails on conditions not met
-                if (exists)
-                {
-                    await blobClient.DeleteAsync(
-                        DeleteSnapshotsOption.IncludeSnapshots,
-                        conditions: _blobClient.GetConditions(fileProperties),
-                        cancellationToken);
-                }
-                else
-                {
-                    _logger.LogInformation(
-                        "Can not delete blob in external store with watermark: '{Version}' and PartitionKey: {PartitionKey}. Dangling SQL Index detected.",
-                        version, partition.Key);
-                }
-
-                return exists;
+                // NOTE - when file does not exist but conditions passed in, it fails on conditions not met
+                return await blobClient.DeleteIfExistsAsync(
+                    DeleteSnapshotsOption.IncludeSnapshots,
+                    conditions: _blobClient.GetConditions(fileProperties),
+                    cancellationToken);
             }
             catch (RequestFailedException ex) when (ex.ErrorCode == BlobErrorCode.ConditionNotMet &&
                                                     _blobClient.IsExternal)
             {
-                // when using external store and condition on etag match not met but blob exists, rethrow
-                throw ex;
+                string message = string.Format(
+                    CultureInfo.InvariantCulture,
+                    DicomCoreResource.ExternalDataStoreOperationFailed,
```

Comment:
Should this be a different resource string, like:

```xml
<data name="ExternalDataStoreBlobModified" xml:space="preserve">
  <value>The blob '{0}' with eTag '{1}' has been modified or deleted.</value>
</data>
```

Is it worth promoting this new message to `ExecuteAsync` where we can reuse this message for things like WADO as well. Shouldn't all of the `Get*` methods in this class attach an eTag condition to their respective API calls?